feat(helm): added values README via helm-docs cli

justmike1 opened the pull request:
### What does this PR do?

<!-- A brief description of the change being made with this pull request. -->

Added README file to possess all values that traefik's helm chart reads

### Motivation

<!-- What inspired you to submit this pull request? -->
I checked the full value path I needed to write and reading from the yaml is difficult, for example `ports` is a long tree.

### More

- [ ] Yes, I updated the tests accordingly
- [ ] Yes, I ran `make test` and all the tests passed

<!--

HOW TO WRITE A GOOD PULL REQUEST? https://doc.traefik.io/traefik/contributing/submitting-pull-requests/

-->

mloiseleur commented:
Hello @justmike1, 

Thanks for your interests in Traefik and for this PR !
This can definitely help when using this chart.

I have some comments:
1. Wdyt about adding a Makefile target to re-generate this file?
2. There are a lot of parameters, so maybe we should keep this doc inside a specific file (like `VALUES.md`), in order to keep readability
3. README.md is copied to `traefik/` in order to be displayed in [artifacthub page](https://artifacthub.io/packages/helm/traefik/traefik/). See [here](https://github.com/traefik/traefik-helm-chart/blob/master/.github/workflows/release.yml#L30).

justmike1 commented:
@mloiseleur 
1. Added the Makefile target
2. You can't change the name of the file the `helm-docs` output. so I added renaming via Makefile
3. Changing to `VALUES.md` mitigates the issue, though in the future, when we will add also proper descriptions and default values via `helm-docs` supported comments and have a descriptive file, we will need to think a way to merge them sensibly and not just one on top of the other.

I use `traefik` for a year already and I don't see it change any time soon, so I will improve it with time.

justmike1 commented:
I also added a test step in order for the file to stay updated upon PRs

mloiseleur commented:
That's better. I tried it by adding the needed `--` into values.yaml for two first section and it looks good. 
I guess we'll need to find how to properly display examples provided directly into values.yaml.

justmike1 commented:
@mloiseleur anything more required to push this PR? I don't like big PRs, so I would suggest steps:
1. merging this PR 
2. PR for refactoring values.yaml comments to `helm-docs`
3. PR for main README to have those values as well with keeping it's structure, like injecting them before `contributions` header, in order to be seen also in artifacthub

mloiseleur commented:
I don't have a strong opinion of PR size.
I do have a strong opinion that we should not merge an empty VALUES.md.
I agree with you that it could be merged within the README.

=> For this PR, I think it's missing two things:
1. Add the needed `--` in values.yaml, to have a base on VALUES.md
2. In [Guidelines](https://github.com/traefik/traefik-helm-chart/blob/master/traefik/Guidelines.md#non-critical-feature-example), non critical features are commented. How could those features be documented with this system ?

justmike1 commented:
@mloiseleur Sorry for repeating commits, `VALUES.md` is full now, pending your next review.
Currently, I don't see a way in `helm-docs` to document commented out parameters, maybe in future versions, so I believe keeping it as is, and maybe we can layer the documentation, and then make a full one that will be structured from the files in CD flow.

mloiseleur commented:
I should be able to review it on next friday. I'll keep you informed.

justmike1 commented:
Thank you guys! @darkweaver87 @mloiseleur
